llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -437,6 +437,13 @@ X86TargetLowering::X86TargetLowering(const X86TargetMachine &TM,
     setOperationAction(Op, MVT::f128, Expand);
   }
 
+  // FIXME: If a target has F16C, it needs to be Custom.
+  setOperationAction(ISD::STRICT_FP_TO_BF16, MVT::f32, Expand);
```

Comment:
Someone with X86 backend expertise needs to change this to `Custom` and implement `case ISD::STRICT_BF16_TO_FP:  return LowerBF16_TO_FP(Op, DAG);` in `LowerOperation` below.

Reply:
The comment is not correct. We don't need to custom for F16C. We cannot do custom at all because we don't have BF16 truncation instrcutions that support exception.
That's why I suggest you add a test for AVX512BF16, e.g.,
```
; RUN: llc < %s -mtriple=x86_64-linux-gnu -mattr=+avx512bf16,+avx512vl | FileCheck %s --check-prefix=X64
```

Reply:
I think we don't need to exclude `f80` and `f128`. We can use a loop for them.
```
for (auto VT : {MVT::f32, MVT::f64, MVT::f80, MVT::f128}) {
  setOperationAction(ISD::STRICT_FP_TO_BF16, VT, Expand);
  setOperationAction(ISD::STRICT_BF16_TO_FP, VT, Expand);
}
```